src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -0,0 +1,299 @@
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/event_engine/event.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+
+namespace am = application_manager;
+using am::commands::CommandImpl;
+using am::ApplicationManager;
+using am::commands::MessageSharedPtr;
+using am::ApplicationSharedPtr;
+using am::MockMessageHelper;
+using ::testing::_;
+using ::utils::SharedPtr;
+using ::testing::Return;
+
+using ::test::components::application_manager_test::MockApplication;
+using usage_statistics_test::MockStatisticsManager;
+using am::event_engine::EventObserver;
+
+namespace smart_objects = NsSmartDeviceLink::NsSmartObjects;
+namespace custom_str = utils::custom_string;
+namespace strings = ::application_manager::strings;
+namespace hmi_response = ::application_manager::hmi_response;
+
+class AddCommandRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  sync_primitives::Lock lock_;
+  MessageSharedPtr create_fullParams_ui_so() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    smart_objects::SmartObject menu_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    menu_params[strings::position] = 10;
+    menu_params[strings::menu_name] = "LG";
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params[strings::cmd_id] = 1;
+    msg_params[strings::menu_params] = menu_params;
+    msg_params[strings::app_id] = 34;
+    msg_params[strings::cmd_icon] = 1;
+    msg_params[strings::cmd_icon][strings::value] = "10";
+    (*msg)[strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  MessageSharedPtr create_fullParams_vr_so() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params[strings::cmd_id] = 2;
+    msg_params[strings::vr_commands] =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+    msg_params[strings::vr_commands][0] = "lamer";
+    msg_params[strings::type] = 34;
+    msg_params[strings::grammar_id] = 12;
+    msg_params[strings::app_id] = 20;
+    (*msg)[strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  MessageSharedPtr create_basic_so() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    smart_objects::SmartObject menu_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    menu_params[strings::position] = 10;
+    menu_params[strings::menu_name] = "LG";
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params[strings::cmd_id] = 1;
+    msg_params[strings::app_id] = 34;
+    (*msg)[strings::msg_params] = msg_params;
+
+    return msg;
+  }
+};
+
+const hmi_apis::FunctionID::eType kInvalidFunctionId_ =
```

Comment:
@wolfylambova22 Wrap the constant in an anonymous namespace to prevent access from other files:
`namespace {`
`const hmi_apis::FunctionID::eType kInvalidFunctionId = hmi_apis::FunctionID::INVALID_ENUM;`
` }  // namespace`